ci/check-documentation.py

```diff
@@ -79,6 +99,17 @@ def complain(message):
                     ' does not exist or is not a file')
                 continue
 
+            if check_md_links and url not in readme_files:
+                readme_files.append(url)
```

Comment:
The fact that you don't want `readme_files` to contain duplicates suggests that you need a set rather than a list.